src/runtime_src/core/tools/common/ReportPlatforms.cpp

```diff
@@ -72,6 +73,10 @@ ReportPlatforms::writeReport( const xrt_core::device* /*_pDevice*/,
     _output << boost::format("  %-23s: %s \n") % "Mig Calibrated" % pt_status.get<std::string>("mig_calibrated");
     _output << boost::format("  %-23s: %s \n") % "P2P Status" % pt_status.get<std::string>("p2p_status");
 
+    boost::optional<const boost::property_tree::ptree&> pt_config = pt_platform.get_child_optional("config.p2p");

```

Comment:
Optional does work here.  Unfortunately, in the past not ALL versions of gcc correctly compiles this statement correctly.  While I don't think we are support those older OSes any more, if this code should happen to fail on other OSes, the following solution works as well:

```
const boost::property_tree::ptree ptEmpty;
const boost::property_tree::ptree pt_& pt_config = pt_platform.get_child("config.p2p", ptEmpty);

if (!pt_config.empty()) {
   ...
```